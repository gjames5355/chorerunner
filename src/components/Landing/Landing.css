/* 
font-family: 'Press Start 2P', cursive;
font-family: 'Fredoka One', cursive;
font-family: 'Concert One', cursive;
font-family: 'Electrolize', sans-serif; */

.landing {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  /* width: 80%;  */
  margin: 0 auto 50px auto;
}

.landing img {
  width: 100%;
}

.greeting {
  position: relative;
  height: 85vh;
  padding-top: 15vh;
  background-image: url('../../ImgAssets/ChoreRunner.png');
  background-size: cover;
  background-position: 0 0;
  background-repeat: repeat;
  animation: house 45s linear infinite;
  width: 100%;
}

.greeting>h1 {
  font-family: 'Press Start 2P', cursive;
  font-size: 2.4em;
  color: rgb(255, 217, 0);
  text-shadow: -1px -1px rgb(105, 90, 0), 4px 4px rgb(134, 112, 0);
  animation: bounce 500ms infinite alternate;
  -webkit-animation: bounce 500ms infinite alternate;
}

.about {
  display: flex;
  flex-direction: column;
  width: 300px;
  margin: auto;
  align-items: center;
}

.about>a {
  padding-bottom: 30px;
  font-family: 'Fredoka One', cursive;
  color: white;
  line-height: 0;
  /* display:none; */
}

.about {
  padding-bottom: 30px;
  font-family: 'Electrolize', sans-serif;
  color: grey;
  animation: bounce 500ms infinite alternate;
  -webkit-animation: bounce 500ms infinite alternate;
}

.down-arrow {
  display: block;
  margin: auto;
  width: 100px;
}

/* .shade {
 position: relative;
  height: 85vh;
  background-color: rgb(0, 0, 0,.5);
  background-size: cover;
width: 100vw;
} */


.tagline {
  text-align: center;
  font-size: 1.2em;
  color: white;
  font-family: 'Fredoka One', cursive;
}

.landing h3 {
  padding-top: 20px;
  color: rgb(255, 217, 0);
  font-size: 20px;
  text-shadow: none;
  font-family: 'Press Start 2P', cursive;
}

h2 {
  font-family: 'Fredoka One', cursive;
  letter-spacing: 2px;
  font-size: 20px;
  font-weight: 300;
  color: #725F00;
  text-shadow: -.5px -.5px rgb(105, 90, 0), .5px .5px white;

}

.header-background {
  margin: 0 auto;
  background: rgb(255, 217, 0);
  width: 120px;
  height: 40px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  border-radius: 15px;
  margin-top: 100px;
}

.landing p {
  font-family: 'Electrolize', sans-serif;
  /* padding: 20px 20px; */
  margin: 15px auto;
  text-align: center;

}

.button-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-bottom: 30px;

}

.button-container a {
  width: 290px;
  text-align: center;
  margin: 10px;
  font-family: 'Fredoka One', cursive;
  font-size: 35px;
  padding: 10px 0 10px 10px;
  background-color: rgb(255, 217, 0);
  text-shadow: -1px -1px rgb(105, 90, 0), 1px 1px white;
  color: #725F00;
  -webkit-border-radius: 7px;
  -moz-border-radius: 7px;
  -o-border-radius: 7px;
  border-radius: 7px;
  box-shadow: 0 .3em rgb(134, 112, 0);
  cursor: pointer;
}

.button-container a:active {
  box-shadow: none;
  position: relative;
  top: .2em;
}

.kid-button,
.parent-button,
.new-button {
  text-align: center;
  margin: 10px;
  font-family: 'Fredoka One', cursive;
  font-size: 35px;
  padding: 10px 0 10px 0px;
  background-color: rgb(255, 217, 0);
  text-shadow: -1px -1px rgb(105, 90, 0), 1px 1px white;
  color: #725F00;
  -webkit-border-radius: 7px;
  -moz-border-radius: 7px;
  -o-border-radius: 7px;
  border-radius: 7px;
  box-shadow: 0 .3em rgb(134, 112, 0);
  cursor: pointer;
  max-width: 290px;
}

.img-placeholder {
  display: none;
}

.kids {
  display: flex;
  flex-direction: column;
}

.kids h4,
.parents h4 {
  color: rgb(255, 217, 0);
  font-size: 25px;
  font-family: 'Electrolize', sans-serif;
}

.kids p,
.parents p {
  color: white;
  line-height: 25px;
  text-align: center;
}

.parents {
  display: flex;
  flex-direction: column;

}

.for-kids,
.for-parents {
  padding-top: 100px;
  width: 80%;
}

@media (min-width: 400px) {

  .kid-button,
  .parent-button,
  .new-button {
    width: 290px;
    margin: 10px auto;

  }

}

@media (min-width: 700px) {

  .button-container {
    padding-top: 10vh;
  }
}

@media (min-width: 800px) {

  .landing>h1 {
    font-size: 40px;
  }

  .parents {
    padding: 10px;
    word-wrap: break-word;
  }

  .img-placeholder {
    display: flex;
    justify-content: center;
    padding: 10px;
    align-items: center;
    width: 50%;
  }

  .badge-placeholder {
    display: flex;
    justify-content: center;
    padding: 10px;
    align-items: center;
    width: 50%;
  }

  .kids-container,
  .parent-container {
    display: flex;

  }
}

@media screen and (min-width: 1100px) {

  .for-kids,
  .for-parents {

    width: 1000px;
  }

  .badge-placeholder {
    width: 40%;
  }

  .kids {
    width: 60%;
  }

  .parents {
    min-width: 330px;
  }

  .img-placeholder img {
    width: 450px;
  }

}