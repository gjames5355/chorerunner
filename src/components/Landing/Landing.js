import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCaretDown } from '@fortawesome/free-solid-svg-icons';
import './Landing.css';
import images from '../../ImgAssets/index';

export default class Landing extends Component {
  render() {
    return (
      <div className='landing'>
        <div className='greeting'>
          <h1>Chore Runner</h1>
          <p className='tagline'>The Smart Shared To-Do List</p>
          <div className='button-container'>
            <Link
              className='kid-button'
              tabIndex={1}
              style={{ textDecoration: 'none' }}
              to='/kidLogin'
            >
              I am a kid
            </Link>
            <Link
              className='parent-button'
              tabIndex={2}
              style={{ textDecoration: 'none' }}
              to='/login'
            >
              I am a parent
            </Link>
            <Link
              className='new-button'
              tabIndex={3}
              style={{ textDecoration: 'none' }}
              to='/register'
            >
              I am new
            </Link>
          </div>
          <div className='about'>
            <a style={{ textDecoration: 'none' }} href={'#about-info'}>
              How does it work?
              <FontAwesomeIcon
                className='down-arrow'
                icon={faCaretDown}
                size='5x'
                color='white'
              />
            </a>
          </div>
        </div>

        <div className='for-kids' id='about-info'>
          <h3>For kids</h3>
          <div className='kids-container'>
            <div className='badge-placeholder'>
              <img src={images.kidScreen} alt='Badge'></img>
            </div>
            <div className='kids'>
              <h4>Complete challenges, get badges!</h4>
              <p>
                Chores are boring, games are fun! The rules are simple: race
                your family members to complete chore-lenges, earn points, and
                get badges. However, you can only earn points after your parents
                have approved your chores. Can you collect all 10 badges and
                become the top chorerunner in your house? Ask your parents to
                help you sign up today!
              </p>

              <Link
                className='kid-button'
                tabIndex={4}
                style={{ textDecoration: 'none' }}
                to='/kidLogin'
              >
                I am a kid
              </Link>
            </div>
          </div>
        </div>
        <div className='for-parents'>
          <h3>For parents</h3>
          <div className='parent-container'>
            <div className='parents'>
              <h4>Make chores easy and fun!</h4>
              <p>
                Who's turn is it to do the dishes? Didn't you ask them to finish
                this two days ago?
              </p>
              <p>
                Sound familiar? We know that managing and delegating household
                chores can be a struggle. Let Chorerunner help!
              </p>
              <p>
                Chorerunner makes chores a fun competition between kids and
                family members. Simply sign up, add household members, assign
                tasks, and let chorerunner do the rest. As members complete
                tasks, they move up the household leaderboard, gain levels, and
                earn badges.
              </p>
              <p>
                Spend less time working and more time playing. Sign in or sign
                up to get started!
              </p>
              <Link
                className='parent-button'
                tabIndex={2}
                style={{ textDecoration: 'none' }}
                to='/login'
              >
                I am a parent
              </Link>
              <Link
                className='new-button'
                tabIndex={3}
                style={{ textDecoration: 'none' }}
                to='/register'
              >
                I am new
              </Link>
            </div>
            <div className='img-placeholder'>
              <img src={images.parentScreen} alt='Badge'></img>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
