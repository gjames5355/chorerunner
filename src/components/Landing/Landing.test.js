import React from 'react';
import ReactDOM from 'react-dom';
import Landing from './Landing';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Landing component testing', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <MemoryRouter>
        <Landing />
      </MemoryRouter>,
      div
    );

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI as expected', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Landing />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders empty given no tabs', () => {
    const wrapper = shallow(<Landing />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should count all <p> tags', () => {
    const wrapper = shallow(<Landing />);
    expect(wrapper.find('p').length).toEqual(6);
  });
});
