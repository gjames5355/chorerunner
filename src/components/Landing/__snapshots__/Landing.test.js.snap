// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing component testing renders UI as expected 1`] = `
<div
  className="landing"
>
  <div
    className="greeting"
  >
    <h1>
      Chore Runner
    </h1>
    <p
      className="tagline"
    >
      The Smart Shared To-Do List
    </p>
    <div
      className="button-container"
    >
      <a
        className="kid-button"
        href="/kidLogin"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
        tabIndex={1}
      >
        I am a kid
      </a>
      <a
        className="parent-button"
        href="/login"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
        tabIndex={2}
      >
        I am a parent
      </a>
      <a
        className="new-button"
        href="/register"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
        tabIndex={3}
      >
        I am new
      </a>
    </div>
    <div
      className="about"
    >
      <a
        href="#about-info"
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        How does it work?
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-caret-down fa-w-10 fa-5x down-arrow"
          color="white"
          data-icon="caret-down"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </a>
    </div>
  </div>
  <div
    className="for-kids"
    id="about-info"
  >
    <h3>
      For kids
    </h3>
    <div
      className="kids-container"
    >
      <div
        className="badge-placeholder"
      >
        <img
          alt="Badge"
          src="kidScreen.png"
        />
      </div>
      <div
        className="kids"
      >
        <h4>
          Complete challenges, get badges!
        </h4>
        <p>
          Chores are boring, games are fun! The rules are simple: race your family members to complete chore-lenges, earn points, and get badges. However, you can only earn points after your parents have approved your chores. Can you collect all 10 badges and become the top chorerunner in your house? Ask your parents to help you sign up today!
        </p>
        <a
          className="kid-button"
          href="/kidLogin"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
          tabIndex={4}
        >
          I am a kid
        </a>
      </div>
    </div>
  </div>
  <div
    className="for-parents"
  >
    <h3>
      For parents
    </h3>
    <div
      className="parent-container"
    >
      <div
        className="parents"
      >
        <h4>
          Make chores easy and fun!
        </h4>
        <p>
          Who's turn is it to do the dishes? Didn't you ask them to finish this two days ago?
        </p>
        <p>
          Sound familiar? We know that managing and delegating household chores can be a struggle. Let Chorerunner help!
        </p>
        <p>
          Chorerunner makes chores a fun competition between kids and family members. Simply sign up, add household members, assign tasks, and let chorerunner do the rest. As members complete tasks, they move up the household leaderboard, gain levels, and earn badges.
        </p>
        <p>
          Spend less time working and more time playing. Sign in or sign up to get started!
        </p>
        <a
          className="parent-button"
          href="/login"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
          tabIndex={2}
        >
          I am a parent
        </a>
        <a
          className="new-button"
          href="/register"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
          tabIndex={3}
        >
          I am new
        </a>
      </div>
      <div
        className="img-placeholder"
      >
        <img
          alt="Badge"
          src="parentScreen.png"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Landing component testing renders empty given no tabs 1`] = `
<div
  className="landing"
>
  <div
    className="greeting"
  >
    <h1>
      Chore Runner
    </h1>
    <p
      className="tagline"
    >
      The Smart Shared To-Do List
    </p>
    <div
      className="button-container"
    >
      <Link
        className="kid-button"
        style={
          Object {
            "textDecoration": "none",
          }
        }
        tabIndex={1}
        to="/kidLogin"
      >
        I am a kid
      </Link>
      <Link
        className="parent-button"
        style={
          Object {
            "textDecoration": "none",
          }
        }
        tabIndex={2}
        to="/login"
      >
        I am a parent
      </Link>
      <Link
        className="new-button"
        style={
          Object {
            "textDecoration": "none",
          }
        }
        tabIndex={3}
        to="/register"
      >
        I am new
      </Link>
    </div>
    <div
      className="about"
    >
      <a
        href="#about-info"
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        How does it work?
        <FontAwesomeIcon
          border={false}
          className="down-arrow"
          color="white"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                320,
                512,
                Array [],
                "f0d7",
                "M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z",
              ],
              "iconName": "caret-down",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="5x"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </a>
    </div>
  </div>
  <div
    className="for-kids"
    id="about-info"
  >
    <h3>
      For kids
    </h3>
    <div
      className="kids-container"
    >
      <div
        className="badge-placeholder"
      >
        <img
          alt="Badge"
          src="kidScreen.png"
        />
      </div>
      <div
        className="kids"
      >
        <h4>
          Complete challenges, get badges!
        </h4>
        <p>
          Chores are boring, games are fun! The rules are simple: race your family members to complete chore-lenges, earn points, and get badges. However, you can only earn points after your parents have approved your chores. Can you collect all 10 badges and become the top chorerunner in your house? Ask your parents to help you sign up today!
        </p>
        <Link
          className="kid-button"
          style={
            Object {
              "textDecoration": "none",
            }
          }
          tabIndex={4}
          to="/kidLogin"
        >
          I am a kid
        </Link>
      </div>
    </div>
  </div>
  <div
    className="for-parents"
  >
    <h3>
      For parents
    </h3>
    <div
      className="parent-container"
    >
      <div
        className="parents"
      >
        <h4>
          Make chores easy and fun!
        </h4>
        <p>
          Who's turn is it to do the dishes? Didn't you ask them to finish this two days ago?
        </p>
        <p>
          Sound familiar? We know that managing and delegating household chores can be a struggle. Let Chorerunner help!
        </p>
        <p>
          Chorerunner makes chores a fun competition between kids and family members. Simply sign up, add household members, assign tasks, and let chorerunner do the rest. As members complete tasks, they move up the household leaderboard, gain levels, and earn badges.
        </p>
        <p>
          Spend less time working and more time playing. Sign in or sign up to get started!
        </p>
        <Link
          className="parent-button"
          style={
            Object {
              "textDecoration": "none",
            }
          }
          tabIndex={2}
          to="/login"
        >
          I am a parent
        </Link>
        <Link
          className="new-button"
          style={
            Object {
              "textDecoration": "none",
            }
          }
          tabIndex={3}
          to="/register"
        >
          I am new
        </Link>
      </div>
      <div
        className="img-placeholder"
      >
        <img
          alt="Badge"
          src="parentScreen.png"
        />
      </div>
    </div>
  </div>
</div>
`;
